src/httpclient.cc

```diff
@@ -0,0 +1,158 @@
+#include "httpclient.h"
+
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+#include "logger.h"
+
+/*****************************************************************************/
+/**
+ * \par Description:
+ *    A writeback handler for the curl library. I handles writing response
+ *    data from curl into a string.
+ *    https://curl.haxx.se/libcurl/c/CURLOPT_WRITEFUNCTION.html
+ *
+ */
+static size_t writeString(void* contents, size_t size, size_t nmemb,
+                          void* userp) {
+  // clear the provided string
+  ((std::string*)userp)->clear();
+
+  // append the writeback data to the provided string
+  ((std::string*)userp)->append((char*)contents, size * nmemb);
+
+  // return size of written data
+  return size * nmemb;
```

Comment:
No, because callback may be called twice for one request, but there was an error in prev line , I will remove string clearing in callback and will do it before request perform.